app/api/validations.py

```diff
@@ -0,0 +1,109 @@
+from flask import request
+from app.models import Resource
+from enum import Enum
+from app.utils import standardize_response
+
+
+def requires_body(func):
+    def wrapper(*args, **kwargs):
+        try:
+            # JSON body is {}
+            if not request.get_json():
+                return missing_json_error()
+        except Exception as e:
+            # JSON body is completely missing
+            if "Expecting value: line 1 column 1" in str(e):
+                return missing_json_error()
+
+        return func(*args, **kwargs)
+    return wrapper
+
+
+def missing_json_error():
+    message = "You must provide a valid JSON body to use this endpoint"
+    error = {'errors': {"missing-body":{"message":message}}}
+    return standardize_response(error, status_code=422)
```

Comment:
TODO: Add all these changes (like all of these breaking changes) to the openapi.yaml file so they make it into the docs